.github/workflows/github_actions.yml

```diff
@@ -0,0 +1,100 @@
+name: TileDB-Spark
+
+on:
+  push:
+    tags:
+      - '*'
+    branches:
+      - 'master'
+  pull_request:
+
+jobs:
+  Ubuntu-test:
+#    if: github.ref == 'refs/heads/master'
+    name: Ubuntu-latest
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: unset SYSTEM; ./gradlew checkFormat;
+        shell: bash
+        name: checkFormat
+      - run: unset SYSTEM; ./gradlew assemble;
+        shell: bash
+        name: assemble
+      - run: unset SYSTEM; ./gradlew test;
+        name: test
+        shell: bash
+
+  MacOS-test:
+#    if: github.ref == 'refs/heads/master'
+    name: macOS
+    runs-on: macos-10.15
+    steps:
+      - uses: actions/checkout@v2
+      - run: unset SYSTEM; ./gradlew checkFormat;
+        shell: bash
+        name: checkFormat
+      - run: unset SYSTEM; ./gradlew assemble;
+        shell: bash
+        name: assemble
+      - run: unset SYSTEM; ./gradlew test;
+        shell: bash
+        name: test
+
+  Release:
+    if: startsWith(github.ref, 'refs/tags/')
+    needs: [Ubuntu-test, MacOS-test]
+    name: Create-Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Build Changelog
+        id: github_release
+        uses: mikepenz/release-changelog-builder-action@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Create-Jars
+        run: unset SYSTEM; set +e; ./gradlew assemble; ./gradlew shadowJar; mkdir jars/; cp ./build/libs/*.jar jars/
+
+      #get release version from branch name
+      - name: Get the version
+        id: get_version
+        run: echo ::set-output name=VERSION::$(echo $GITHUB_REF | cut -d / -f 3)
+
+      - name: Create Release
+        id: create_release
+        uses: softprops/action-gh-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.MY_PERSONAL_ACCESS_TOKEN }}
```

Comment:
Why a different token here?

Reply:
Fixed. No need to have a different token, it was a leftover.